waku/v2/node/rest/store/handlers.nim

```diff
@@ -0,0 +1,251 @@
+proc makeHistoryQuery(selfNode: WakuNode,
+                      histQuery: HistoryQuery,
+                      storePeer: RemotePeerInfo):
+
+                      Future[RestApiResponse] {.async.} =
+
+  let queryFut = selfNode.query(histQuery, storePeer)
+  if not await queryFut.withTimeout(futTimeout):
+    const msg = "No history response received (timeout)"
+    error msg
+    return RestApiResponse.internalServerError(msg)
+
+  let res = queryFut.read()
+  if res.isErr():
+    const msg = "Error occurred in queryFut.read()"
+    error msg, error=res.error
+    return RestApiResponse.internalServerError(msg)
+
+  let storeResp = res.value.toStoreResponseRest()
+  let resp = RestApiResponse.jsonResponse(storeResp, status=Http200)
+  if resp.isErr():
+    const msg = "Error building the json respose"
+    error msg, error=resp.error
+    return RestApiResponse.internalServerError(msg)
+
+  return resp.get()
+
+# Converts timestamp in string format to Timestamp object
+proc toTimestamp(input: string): Timestamp =
+  try:
+    return Timestamp(parseInt(input))
+  except CatchableError as e:
+    error "Error converting to Timestamp ", error=e.msg
+    return Timestamp(0)
+
+# Checks whether the string represents a valid positive Timestamp
+proc isValidTimestamp(self: Option[string]): bool =
+  if self.isSome() and self.get() != "":
+    try:
+      let time = parseInt(self.get())
+      return time > 0
+    except CatchableError as e:
+      debug "Problem in isValidTimestamp ", error=e.msg
+
+  return false
+
+# Checks whether the peerAddr parameter represents a valid p2p multiaddress.
+# The param must be in the format `(ip4|ip6)/tcp/p2p/$peerId` but URL-encoded
+proc parsePeerAddr(peerAddr: string): Result[RemotePeerInfo, string] =
+  let parsedAddr = decodeUrl(peerAddr)
+  debug "parsePeerAddr ", parsedAddr
+
+  var storenode: RemotePeerInfo = nil
+  try:
+    storenode = parseRemotePeerInfo(parsedAddr)
+  except Exception as e:
+    let msg = "Failed parsing remote peer info [" & e.msg & "]"
+    return err(msg)
+
+  return ok(storenode)
+
+# Creates a HistoryQuery from the given params
+proc createHistoryQuery(pubsubTopic: Option[string],
+                        contentTopics: Option[string],
+                        senderTime: Option[string],
+                        storeTime: Option[string],
+                        digest: Option[string],
+                        startTime: Option[string],
+                        endTime: Option[string],
+                        pageSize: Option[string],
+                        ascending: Option[string]):
+
+                  Result[HistoryQuery, string] =
+
+  # Parse pubsubTopic parameter
+  var parsedPubsubTopic = none(string)
+  if pubsubTopic.isSome():
+    let decodedPubsubTopic = decodeUrl(pubsubTopic.get())
+    if decodedPubsubTopic != "":
+      parsedPubsubTopic = some(decodedPubsubTopic)
+    debug "Setting pubsubTopic ", pubsub_topic=parsedPubsubTopic.get()
```

Comment:
We need to be consistent with the logged fields name style: sometimes you use the `snake_case`; others you use `camelCase` (e.g., `parsedAddr`). IMO, we should stick with `snake_case`.